#11 Validations on SignUp Screen

OrlandoAlPe opened the pull request:
Completed the following validations for the SignUp Screen:

- Name TextInput
- Email TextInput
- Password TextInpt
- Term and policy agreement checkbox

The app verifies if those validations are passed or not with an alert message.

![Screenshot_1618513521](https://user-images.githubusercontent.com/33968113/114924596-c86afa80-9df3-11eb-9bbc-cc3d0817703a.png)

![Screenshot_1618513683](https://user-images.githubusercontent.com/33968113/114924846-049e5b00-9df4-11eb-8dd1-9c649acd30b1.png)


Firebase SignUp services are Still pending.

OrlandoAlPe committed:
#11 Validations

Completed the following validations for the SignUp Screen:

- Name TextInput
- Email TextInput
- Password TextInpt
- Term and policy agreement checkbox

jkzasori reviewed (commented):
Are you using codeclima? I'm asking because I can't see at the bottom...

Coreandrum1 commented:
> Are you using codeclima? I'm asking because I can't see at the bottom...

OrlandoAlPe commented:
> Are you using codeclima? I'm asking because I can't see at the bottom...
The repostory is private so we can´t implement code climate yet.

francisco-sle reviewed (commented):
LGTM

OrlandoAlPe committed:
Merge branch 'sand-box' into #11-SignUp-screen-validations

Joaquin-Niembro commented on reservacionVuelo/Screens/SignUp.js:
```diff
@@ -1,59 +1,138 @@
+import {CustomCheckbox} from '../components/checkbox';
+import {CustomTextInput} from '../components/TextInput';
+
+const emailRegex = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
+const nameRegex = /^[A-Za-z]+$/;
```
Put this expressions in a constants file

Joaquin-Niembro commented on reservacionVuelo/Screens/SignUp.js:
```diff
@@ -1,59 +1,138 @@
+            })
+          }
+          checkError={this.state.nameStatus === false}
+          errorMessage={'Write a valid name please'}
```
Put the errorMessage in a constants file

Joaquin-Niembro commented on reservacionVuelo/Screens/SignUp.js:
```diff
@@ -1,59 +1,138 @@
+        <CustomCheckbox
+          value={this.state.checkBox}
+          onValueChange={() => this.setState({checkBox: !this.state.checkBox})}
+          text={'I agree to the '}
+          labelText={'Terms'}
+          secondaryText={' and '}
+          secondaryLabelText={'Privacy Policy.'}
+          check={this.state.checkBox}
+          errorMessage={'Terms and privacy policy must be checked'}
```
Put these messages and texts in a constants file

Joaquin-Niembro commented on reservacionVuelo/components/checkbox.js:
```diff
@@ -0,0 +1,38 @@
+import styles from './styles';
+
+export class CustomCheckbox extends Component {
+  constructor() {
+    super();
+  }
```
Remove the constructor if its not in use

OrlandoAlPe committed:
#11 CR FIX

Created Constants file
moved error and description messages to constants file, moved regex constants too.
Removed the unused constructor from custom checkbox and custom text input components.

DiegoSpielzeug commented:
why don't you have CodeClimate and Deepscan?
